src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private T[] grow(int oldSize) {
+        int newSize = oldSize + oldSize / 2;
+        T[] newElementData = (T[]) new Object[newSize];
+        System.arraycopy(elementData, 0, newElementData, 0, elementData.length);
+        return newElementData;
+    }
+
+    public void checkNonExistentOrNegativePosition(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element on this position");
+        }
+    }
+
+    public void checkIndexForAddElement(int index) {
```

Comment:
this should be private
if it's used only once, you can, but don't need to extract the logic

Reply:
OK